### ISSUE TITLE
Only 5 retries instead of 20...

### ISSUE THREAD

**Issue by StefanoGiu** [amitfin/retry#34]
### System Health details

                 - service: retry.call
                   data:
                     service: telegram_bot.send_video
                     retries: 20
                     target: XXX
                     title: XXX
                     message: XXX
                     timeout: XXX
                     url: XXX
                     caption: XXX

In the log:
2023-10-23 13:18:58.147 WARNING (SyncWorker_0) [homeassistant.components.telegram_bot] Status code 414 (retry #1) loading http:\\xxx
2023-10-23 13:18:58.162 WARNING (SyncWorker_0) [homeassistant.components.telegram_bot] Status code 414 (retry #2) loading http:\\xxx
2023-10-23 13:18:58.194 WARNING (SyncWorker_0) [homeassistant.components.telegram_bot] Status code 414 (retry #3) loading http:\\xxx
2023-10-23 13:18:58.209 WARNING (SyncWorker_0) [homeassistant.components.telegram_bot] Status code 414 (retry #4) loading http:\\xxx
2023-10-23 13:18:58.231 WARNING (SyncWorker_0) [homeassistant.components.telegram_bot] Status code 414 (retry #5) loading http:\\xxx
2023-10-23 13:18:58.231 WARNING (SyncWorker_0) [homeassistant.components.telegram_bot] Can't load data in http:\\xxx after 5 retries

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of currently [previous issues](https://github.com/amitfin/retry/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

I set 20 retries, but it tries only 5 times.

### Reproduction steps

Use retries: 20

### Debug logs

```text
2023-10-23 13:18:58.147 WARNING (SyncWorker_0) [homeassistant.components.telegram_bot] Status code 414 (retry #1) loading http:\\xxx
2023-10-23 13:18:58.162 WARNING (SyncWorker_0) [homeassistant.components.telegram_bot] Status code 414 (retry #2) loading http:\\xxx
2023-10-23 13:18:58.194 WARNING (SyncWorker_0) [homeassistant.components.telegram_bot] Status code 414 (retry #3) loading http:\\xxx
2023-10-23 13:18:58.209 WARNING (SyncWorker_0) [homeassistant.components.telegram_bot] Status code 414 (retry #4) loading http:\\xxx
2023-10-23 13:18:58.231 WARNING (SyncWorker_0) [homeassistant.components.telegram_bot] Status code 414 (retry #5) loading http:\\xxx
2023-10-23 13:18:58.231 WARNING (SyncWorker_0) [homeassistant.components.telegram_bot] Can't load data in http:\\xxx after 5 retries
```

**Comment by amitfin** [comment]
The warning log entries listed above are from the `telegram_bot` integration itself ([here](https://github.com/home-assistant/core/blob/54bcd70878668744d7eccabfc81f40a19117e38a/homeassistant/components/telegram_bot/__init__.py#L313)), and are unrelated to the `retry` integration. The 5 retries is coming from [here](https://github.com/home-assistant/core/blob/54bcd70878668744d7eccabfc81f40a19117e38a/homeassistant/components/telegram_bot/__init__.py#L292).
The `telegram_bot.send_video` service runs its logic in the background. The background tasks is created [here](https://github.com/home-assistant/core/blob/54bcd70878668744d7eccabfc81f40a19117e38a/homeassistant/components/telegram_bot/__init__.py#L418). So, from the `retry.call` perspective the 1st service call succeeded and there is no need to call it again.
Bottom line: `retry.call` can't be used in this scenario as the inner service call always returns successfully, while running the actual logic in the background.
